src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/Contents.java

```diff
@@ -407,10 +407,10 @@ public Content getContent(String key, Object o0, Object o1, Object o2) {
 
             if (o == null) {
                 c.add("{" + m.group(1) + "}");
-            } else if (o instanceof String) {
-                c.add((String) o);
-            } else if (o instanceof Content) {
-                c.add((Content) o);
+            } else if (o instanceof String str) {
+                c.add(str);
+            } else if (o instanceof Content con) {
+                c.add(con);
             }
 
```

Comment:
I guess at some point this becomes "pattern switch" or whatever the feature is called ... but not yet

Reply:
The feature you are talking about is called "Pattern Matching for switch". The feature is described as a preview feature by JEP 406, which is currently proposed to target JDK 17.